### ISSUE TITLE
`poetry version` does not work when [[tool.poetry.source]] in pyproject.toml

### ISSUE THREAD

**Issue by Sweenu** [sdispater/poetry#753]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS Mojave
- **Poetry version**: 0.12.10
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

<details>
<summary>pyproject.toml</summary>

<pre>
[[tool.poetry.source]]
name = "myindex"
url = "https://pypi.mydomain.com/simple/"

[tool.poetry]
name = "myproject"
version = "0.1.0"
description = ""
authors = ["Bruno Inec <bruno@inec.fr>"]

[tool.poetry.dependencies]
python = "^2.7"

[tool.poetry.dev-dependencies]
pytest = "^3.0"

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
</pre>
</details>

## Issue

`poetry version` does not actually write the new version to file when a `[[tool.poetry.source]]` is present in the `pyproject.toml`.

It is probably a bug in `tomlkit` because in the following code (`console/commands/version.py`):
```python
content = self.poetry.file.read()
poetry_content = content["tool"]["poetry"]
poetry_content["version"] = version.text

self.poetry.file.write(content)
```
`content` is not actually changed when `source: ['name': 'myindex', 'url': 'https://pypi.mydomain.com/simple/']` is present in `poetry_content` while it _is_ changed if not.

I suggest adding some integration tests to the `version` command to check that the new version is actually written in the pyproject.toml so we can catch these types of errors.

**Comment by seifertm** [comment]
The issue seems to be gone since v0.12.11. Can you check on your side?

**Comment by Sweenu** [comment]
It is fixed in v0.12.11! :)
